pkg/agent/openflow/pipeline.go

```diff
@@ -952,34 +966,65 @@ func (c *client) conjunctionActionFlow(conjunctionID uint32, tableID binding.Tab
-		LoadToLabelRange(uint64(conjunctionID), &labelRange).
-		CTDone().
-		Cookie(c.cookieAllocator.Request(cookie.Policy).Raw()).
-		Done()
+	if enableLogging {
+		return c.pipeline[tableID].BuildFlow(ofPriority).MatchProtocol(binding.ProtocolIP).
+			MatchConjID(conjunctionID).
+			MatchPriority(ofPriority).
+			Action().LoadRegRange(int(conjReg), conjunctionID, binding.Range{0, 31}).       // Traceflow.
+			Action().LoadRegRange(int(marksReg), DispositionAllow, APDispositionMarkRange). //AntreaPolicy
+			Action().SendToController(uint8(PacketInReasonNP)).
+			Action().CT(true, nextTable, CtZone).
+			LoadToLabelRange(uint64(conjunctionID), &labelRange).
+			CTDone().
+			Cookie(c.cookieAllocator.Request(cookie.Policy).Raw()).
+			Done()
+	} else {
+		return c.pipeline[tableID].BuildFlow(ofPriority).MatchProtocol(binding.ProtocolIP).
+			MatchConjID(conjunctionID).
+			MatchPriority(ofPriority).
+			Action().LoadRegRange(int(conjReg), conjunctionID, binding.Range{0, 31}). // Traceflow.
+			Action().CT(true, nextTable, CtZone).                                     // CT action requires commit flag if actions other than NAT without arguments are specified.
+			LoadToLabelRange(uint64(conjunctionID), &labelRange).
+			CTDone().
+			Cookie(c.cookieAllocator.Request(cookie.Policy).Raw()).
+			Done()
+	}
 }
 
 // conjunctionActionDropFlow generates the flow to mark the packet to be dropped if policyRuleConjunction ID is matched.
 // Any matched flow will be dropped in corresponding metric tables.
-func (c *client) conjunctionActionDropFlow(conjunctionID uint32, tableID binding.TableIDType, priority *uint16) binding.Flow {
+func (c *client) conjunctionActionDropFlow(conjunctionID uint32, tableID binding.TableIDType, priority *uint16, enableLogging bool) binding.Flow {
 	ofPriority := *priority
+	conjReg := IngressReg
 	metricTableID := IngressMetricTable
 	if _, ok := egressTables[tableID]; ok {
+		conjReg = EgressReg
 		metricTableID = EgressMetricTable
 	}
 	// We do not drop the packet immediately but send the packet to the metric table to update the rule metrics.
-	return c.pipeline[tableID].BuildFlow(ofPriority).MatchProtocol(binding.ProtocolIP).
-		MatchConjID(conjunctionID).
-		MatchPriority(ofPriority).
-		Action().LoadRegRange(int(cnpDropConjunctionIDReg), conjunctionID, binding.Range{0, 31}).
-		Action().LoadRegRange(int(marksReg), cnpDropMark, cnpDropMarkRange).
-		Action().GotoTable(metricTableID).
-		Cookie(c.cookieAllocator.Request(cookie.Policy).Raw()).
-		Done()
+	if enableLogging {
+		return c.pipeline[tableID].BuildFlow(ofPriority).MatchProtocol(binding.ProtocolIP).
+			MatchConjID(conjunctionID).
+			MatchPriority(ofPriority).
+			Action().LoadRegRange(int(cnpDropConjunctionIDReg), conjunctionID, binding.Range{0, 31}).
+			Action().LoadRegRange(int(conjReg), conjunctionID, binding.Range{0, 31}). // Logging
```

Comment:
The conjunctionID will be stored in two regs for different purposes? Is it possible to reuse the reg?

Reply:
Consistently used CNPDropConjunctionIDReg and checked disposition to determine which reg to unload.